### PR TITLE
CT-3232 Create the complaint case from offender SAR

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to
1 - Add the feature of raising the complaint from Offender SAR case
2 - Add a stamp message when a offender SAR links to a complaint case
3 - Added some initial states for complaint workflow

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3232
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3259
### Manual testing instructions

1 - Login as Branson user
2 - Choose.a closed offender sar case
3 - should see a button named as 'Start complaint'
4 - Follow the steps to create a complaint case
5 - once the steps is done, a stamp message should appear in the history of linked offender sar case
5 - The initial state of the complaint case is "To be assessed"

### CODE REVIEW

File: app/state_machines/configurable_state_machine/machine.rb [hunk 1]
```diff
@@ -21,12 +21,15 @@ def self.states
...
-       "ready_to_dispatch",
+       'data_to_be_requested', 
+       'waiting_for_data', 
+       'ready_for_vetting', 
+       'vetting_in_progress', 
+       'ready_to_copy', 
+       'ready_to_dispatch',
+       'to_be_assessed',
+       'data_review_to_be_required',
+       'response_to_be_required',
```
Comment (Polsonby) [inline app/state_machines/configurable_state_machine/machine.rb:32]: wondering about the naming here - how about:

```suggestion
       'data_review_required',
       'response_required', 
```

File: app/state_machines/workflows/predicates.rb [hunk 2]
```diff
@@ -139,6 +139,9 @@ def assigned_team_member_and_case_outside_escalation_period?
     responder_is_member_of_assigned_team? && @kase.outside_escalation_deadline?
   end
 
+  def can_start_to_complain?
+    @kase.offender_sar? && (@kase.already_late? || @kase.current_state == 'closed')
+  end
```
Comment (Polsonby) [inline app/state_machines/workflows/predicates.rb:144]: Are we sure you can't have a complaint about a case before it's "late"?

File: config/locales/state.en.yml [hunk 3]
```diff
@@ -23,6 +23,9 @@ en:
       ready_to_dispatch: Ready to dispatch
       vetting_in_progress: Vetting in progress
       waiting_for_data: Waiting for data
+      to_be_assessed: To be assessed
+      data_review_to_be_required: Date review is required
```
Comment (Polsonby) [inline config/locales/state.en.yml:27]: Typo? 
```suggestion
      data_review_to_be_required: Data review is required
```

File: app/policies/case/sar/offender_policy.rb [hunk 4]
```diff
@@ -55,6 +55,15 @@ def can_send_dispatch_letter?
     check_user_can_manage_offender_sar
   end
 
+  def can_start_to_complain?
+    clear_failed_checks
+    check_user_can_manage_offender_complaint
+  end
```
Comment (Polsonby) [inline app/policies/case/sar/offender_policy.rb:61]: Could we maybe call this `can_start_complaint`?

File: db/seeders/correspondence_type_seeder.rb [hunk 5]
```diff
@@ -32,28 +37,63 @@ def seed!
...
+                report_category_name: 'Offender SAR Complaint', 
                 escalation_time_limit: 3,
                 internal_time_limit: 10,
                 external_time_limit: 1,
+                extension_time_limit: 2, 
+                extension_time_default: 1, 
                 deadline_calculator_class: 'CalendarMonths')
   end
   #rubocop:enable Metrics/MethodLength
+  #rubocop:enable Metrics/CyclomaticComplexity
```
Comment (Polsonby) [inline db/seeders/correspondence_type_seeder.rb:98]: LOL 😆 I was just looking at this, realising we didn't have them in the seeders. Nice one 🚀

File: app/state_machines/workflows/predicates.rb (hunk 2 above)
Comment (ymao2) [reply app/state_machines/workflows/predicates.rb:144]: The rules are based on the discussion in that ticket :) as Alex was thinking whether we should only add this button for closed cases, then recalled Users may raise complaint against open cases and the situation we can think of is due to the timeliness (late),  but actually I don't mind to change it if we get the feedback from users, it is small task anyway

**Review by Polsonby — APPROVED** [review]
Great work :) LGTM

**Pull request merged**